clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -1464,6 +1467,21 @@ class AnnotatingParser {
     }
   }
 
+  void parseEmbedDirective() {
+    if (CurrentToken && CurrentToken->is(tok::less)) {
+      next();
+      while (CurrentToken) {
+        // Mark tokens up to the trailing line comments as implicit string
+        // literals.
+        if (CurrentToken->isNot(tok::comment) &&
+            !CurrentToken->TokenText.startswith("//")) {
+          CurrentToken->setType(TT_ImplicitStringLiteral);
```

Comment:
Can you please add tests in `TokenAnnotatorTest.cpp`?

Reply:
just fyi

> I have no intention of following up on this PR. I am too tired to carry it to fruition. Pick this apart, take from it what you want, or reimplement it entirely. (I will be unsubscribing from this immediately after posting.)

so the "you" you are addressing might indeed be *you*.

Reply:
I started adding the tests and I don't think any of this code is correct. It's basically treating everything in the embed expression as though it was a string literal and that's just plain not how it works. (It also isn't hooked up to do anything useful for `#embed`, just `__has_embed`.)

I'm actually thinking of pulling all of the changes out of the Format directory and letting the clang-format folks do this the correct way after the initial offering lands (similar to how we're not exposing anything to libclang yet). Is that reasonable? CC @mydeveloperday @owenca

Reply:
+1.

Reply:
Thanks! When you get to the point of working on this, I'm happy to review/help how I can.